pwndbg/gdblib/disasm/riscv.py

```diff
@@ -28,15 +29,13 @@ def _is_condition_taken(
         else:
             src2_unsigned = 0
 
-        if self.architecture == "rv32":
-            src1_signed = src1_unsigned - ((src1_unsigned & 0x80000000) << 1)
-            src2_signed = src2_unsigned - ((src2_unsigned & 0x80000000) << 1)
-        elif self.architecture == "rv64":
-            src1_signed = src1_unsigned - ((src1_unsigned & 0x80000000_00000000) << 1)
-            src2_signed = src2_unsigned - ((src2_unsigned & 0x80000000_00000000) << 1)
-        else:
+        if self.architecture not in {"rv32", "rv64"}:
+            # TODO: figure out what this check is for. Are there other riscv arch types in GDB?
             raise NotImplementedError(f"architecture '{self.architecture}' not implemented")
 
+        src1_signed = bit_math.to_signed(src1_unsigned, pwndbg.gdblib.arch.ptrsize * 8)
+        src2_signed = bit_math.to_signed(src2_unsigned, pwndbg.gdblib.arch.ptrsize * 8)
```

Comment:
Can u move the `ptrsize * 8` calculation to a variable?

Also, we probably should have sth like `pwndbg.gdblib.arch.ptrbytes`, heh:
```
(.venv) root@pwndbg:~/pwndbg# rg 'ptrsize \* 8'
pwndbg/gdblib/memory.py
272:        size = pwndbg.gdblib.arch.ptrsize * 8

pwndbg/gdblib/onegadget.py
139:                bits = pwndbg.gdblib.arch.ptrsize * 8

pwndbg/gdblib/arch.py
89:    pwnlib.context.context.bits = ptrsize * 8

pwndbg/commands/dumpargs.py
27:            f"Detected ABI: {pwndbg.gdblib.arch.name} ({pwndbg.gdblib.arch.ptrsize * 8} bit) either doesn't pass arguments through registers or is not implemented. Maybe they are passed on the stack?"
```

Reply:
(In case you want to add `ptrbytes`, pls do so in a different PR)

Reply:
I'll make a note to add this when refactoring our Architecture class.